ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionCount.java

```diff
@@ -45,14 +62,56 @@ public String getName() {
 
   @Override
   public JsonRpcResponse response(final JsonRpcRequest request) {
-    if (request.getParamLength() != 2) {
+    if (request.getParamLength() == 2) {
+      final Address address = parameters.required(request.getParams(), 0, Address.class);
+      final String privacyGroupId = parameters.required(request.getParams(), 1, String.class);
+
+      final long nonce = privateTransactionHandler.getSenderNonce(address, privacyGroupId);
+      return new JsonRpcSuccessResponse(request.getId(), Quantity.create(nonce));
+    } else if (request.getParamLength() == 3) {
+      final Address address = parameters.required(request.getParams(), 0, Address.class);
+      final String privateFrom = parameters.required(request.getParams(), 1, String.class);
+      final String[] privateFor = parameters.required(request.getParams(), 2, String[].class);
+
+      try {
+        final long nonce = determineNonce(privateFrom, privateFor, address);
+        return new JsonRpcSuccessResponse(request.getId(), Quantity.create(nonce));
+      } catch (final Exception e) {
+        LOG.error(e.getMessage(), e);
+        return new JsonRpcErrorResponse(
+            request.getId(), JsonRpcError.GET_PRIVATE_TRANSACTION_NONCE_ERROR);
+      }
+    } else {
       return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
     }
+  }
+
+  private long determineNonce(
```

Comment:
This would be simpler if the EEaPrivateNonceProvider could be used

Reply:
Ups, I removed that class, because it is no longer really determining the nonce, only the privacy group id. I just forgot to do the last bit of it. Done that now and will push it up shortly. Let me know what you think!